backend/app/services/shared_task.py

```
@@ -824,18 +824,27 @@ def get_public_shared_task(
             users = db.query(User).filter(User.id.in_(sender_ids)).all()
             user_name_map = {u.id: u.user_name for u in users}
 
-        # Convert to public subtask data (exclude sensitive fields)
-        public_subtasks = []
-        for sub in subtasks:
-            # Knowledge base contexts are runtime metadata and are not public.
+        contexts_by_subtask_id: Dict[int, List[SubtaskContext]] = {}
```

Ruff (0.15.21)

[error] Undefined name `Dict`

(F821)